src/components/navigationParts/Search.jsx

```diff
@@ -1,34 +1,57 @@
 import { CiSearch } from 'react-icons/ci';
 import { AppContext } from '../../store/AppContext';
-import { useContext, useRef } from 'react';
+import { useEffect, useContext, useRef, useState } from 'react';
 
 const Search = () => {
   const { searchFieldVisible, setSearchFieldVisible, setSelectedLocation } =
     useContext(AppContext);
-
   const inputRef = useRef(null);
+  const [searchHistory, setSearchHistory] = useState([]);
+
+  useEffect(() => {
+    // Load search history from local storage when the component mounts
+    const storedHistory = localStorage.getItem('searchHistory');
+    if (storedHistory) {
+      setSearchHistory(JSON.parse(storedHistory));
+    }
+  }, []);
 
   function handleSearchSubmit(evt) {
     evt.preventDefault();
     const inputValue = inputRef.current.value;
+    const newHistory = [inputValue, ...searchHistory].slice(0, 5);
```

Comment:
why do you put new search at the start of arr?

Reply:
I think that if restored history have length == 5 you'll not be able to save a new one

Reply:
@nkv843 I add the new search result to beginning of the array and slice off the end of array after the 5th elmt. Works like magic.